### ISSUE TITLE
support both DSN and URI when using sql.Open

### ISSUE THREAD

**Issue by james-lawrence** [jackc/pgx#254]
currently sql.Open when using pgx requires using a URI.

It'd be nice if it also supported both DSN or URI. I'm happy to write the PR/Tests if the change would be acceptable.

this would effectively be the required changes:
adding:
```
// ParseURIOrDSN parses URI if the connection string begins with postgres:// or postgresql://
// otherwise fallsback to DSN.
func ParseURIOrDSN(s string) (ConnConfig, error) {
	if strings.HasPrefix(s, "postgres://") || strings.HasPrefix(s, "postgresql://") {
		return ParseURI(s)
	}
	return ParseDSN(s)
}
```

changing func (d *Driver) Open(name string) (driver.Conn, error):
```
	connConfig, err := pgx.ParseURI(name)
	if err != nil {
		return nil, err
	}
```
to
```
	connConfig, err := pgx.ParseURIOrDSN(name)
	if err != nil {
		return nil, err
	}
```

**Comment by james-lawrence** [comment]
Side note: pgx contains an undocumented memory leak when using OpenFromConnPool.

**Comment by jackc** [comment]
Seems like a reasonable change. `ParseURIOrDSN` is a rather awkward name though. Not sure what would be better though... maybe `ParseConnectionString`.

**Comment by jackc** [comment]
Resolved in ef857b600f23bae07b1ada06fde50ef09aeaf7d3
